### PR TITLE
Display caps for vaCopy

### PR DESCRIPTION
**Pull request by XinfengZhang**
https://github.com/intel/libva/pull/513

### CODE REVIEW

File: media_driver/linux/gen12/ddi/media_libva_caps_g12.cpp [hunk 1]
```diff
@@ -2536,6 +2536,49 @@ GMM_RESOURCE_FORMAT MediaLibvaCapsG12::ConvertMediaFmtToGmmFmt(
     }
 }
 
+VAStatus MediaLibvaCapsG12::QueryDisplayAttributes(
+            VADisplayAttribute *attribList,
+            int32_t *numAttribs)
+{
+    DDI_CHK_NULL(attribList, "Null num_attribs", VA_STATUS_ERROR_INVALID_PARAMETER);
+    DDI_CHK_LARGER(*numAttribs,DDI_CODEC_GEN_MAX_DISPLAY_ATTRIBUTES , "Not enough num of attributes", VA_STATUS_ERROR_INVALID_PARAMETER);
```
Comment (dvrogozh) [inline media_driver/linux/gen12/ddi/media_libva_caps_g12.cpp:2544]: `DDI_CHK_NULL(numAttribs)` to avoid crash

**Review by dvrogozh — CHANGES_REQUESTED** [review]

**Review by dmitryermilov — APPROVED** [review]

**Review by Xiaogangli-intel — APPROVED** [review]

**Comment by ceyusa** [comment]
ping?

libva 1.12  is out  with this https://github.com/intel/libva/pull/513 merged, but without support in media-driver :)

File: media_driver/linux/common/ddi/media_libva.h [hunk 2]
```diff
@@ -61,7 +61,7 @@
 #if VA_MAJOR_VERSION < 1
 #define DDI_CODEC_GEN_MAX_DISPLAY_ATTRIBUTES       4
 #else
-#define DDI_CODEC_GEN_MAX_DISPLAY_ATTRIBUTES       0    // set it to zero, unsupported.
+#define DDI_CODEC_GEN_MAX_DISPLAY_ATTRIBUTES       1
```
Comment (fenhu) [inline media_driver/linux/common/ddi/media_libva.h:64]: How about change to DDI_MEDIA_GEN_MAX_DISPLAY_ATTRIBUTES

File: media_driver/linux/common/ddi/media_libva_caps.cpp [hunk 3]
```diff
@@ -3073,6 +3073,55 @@ VAStatus MediaLibvaCaps::QuerySurfaceAttributes(
...
+                attribList->value = VA_ATTRIB_NOT_SUPPORTED;
+                attribList->flags = VA_DISPLAY_ATTRIB_NOT_SUPPORTED;
+                break;
+        }
+        attribList ++;
+    }
+    return VA_STATUS_SUCCESS;
+}
+
+VAStatus MediaLibvaCaps::SetDisplayAttributes(
```
Comment (fenhu) [inline media_driver/linux/common/ddi/media_libva_caps.cpp:3116]: pls remove it if media doesn't need it.

File: media_driver/linux/gen12/ddi/media_libva_caps_g12.cpp [hunk 4]
```diff
@@ -2599,6 +2599,33 @@ GMM_RESOURCE_FORMAT MediaLibvaCapsG12::ConvertMediaFmtToGmmFmt(
...
+VAStatus MediaLibvaCapsG12::GetDisplayAttributes(
+            VADisplayAttribute *attribList,
+            int32_t numAttribs)
+{
+    DDI_CHK_NULL(attribList, "Null attribList", VA_STATUS_ERROR_INVALID_PARAMETER);
+    for(auto i = 0; i < numAttribs; i ++)
+    {
+        switch(attribList->type)
+        {
+            case VADisplayAttribSubDevice:
```
Comment (fenhu) [inline media_driver/linux/gen12/ddi/media_libva_caps_g12.cpp:2611]: Does it for VADisplayAttribCopy?

File: media_driver/linux/common/ddi/media_libva.h (hunk 2 above)
Comment (XinfengZhang) [reply media_driver/linux/common/ddi/media_libva.h:64]: accept

**Review by fenhu — APPROVED** [review]

**Pull request merged**